### PR TITLE
[7964] Iteration on test results API specs to cover the LEC test deta…

### PR DESCRIPTION
**Pull request by jgimber-deloitte**
…ils fields

Added new tests to cover creation / retrieval / verification of the new LEC data fields.

### CODE REVIEW

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+import net.thucydides.core.annotations.WithTags;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import steps.TestResultsSteps;
+import util.JsonPathAlteration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@WithTags(
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: you can erase this as well, it will no longer be used

**Review by bfloreaDVSA — DISMISSED** [review]

**Review by dpanzaru-deloitte — DISMISSED** [review]

**Review by dpanzaru-deloitte — APPROVED** [review]

**Review by dpanzaru-deloitte — DISMISSED** [review]

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 1 above)
Comment (jgimber-deloitte) [reply src/test/java/testresults/PostTestResultsLEC.java]: Removed.

**Comment by dpanzaru-deloitte** [comment]
@jgimber-deloitte Please resolve the conflicts before we approve the PR

**Review by bfloreaDVSA — DISMISSED** [review]

**Review by dpanzaru-deloitte — DISMISSED** [review]

**Review by bfloreaDVSA — DISMISSED** [review]

**Review by dpanzaru-deloitte — DISMISSED** [review]

**Review by dpanzaru-deloitte — DISMISSED** [review]

**Review by bfloreaDVSA — DISMISSED** [review]

File: src/test/java/testnumber/PostTestNumber.java [hunk 2]
```diff
@@ -227,6 +256,7 @@ public void validTestNumberGenerationForAtLeastTwoTestTypes(){
         testResultsSteps.validateTestNumberIsDifferentForTwoTestTypes();
     }
 
+    @WithTag("failing due to CVSB-7964 - to update")
```
Comment (bfloreaDVSA) [inline src/test/java/testnumber/PostTestNumber.java]: no longer required

File: src/test/java/testresults/PostTestResultsNegTestTypesCancelledLvl.java [hunk 3]
```diff
@@ -108,7 +110,7 @@ public void testResultsNullCertificateLink() {
         testResultsSteps.validatePostErrorData("certificateLink", "is not allowed");
     }
 
-
+    @Ignore("blocked by CVSB-9639")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsNegTestTypesCancelledLvl.java]: defect was fixed, please remove the @Ignore annotation

File: src/test/java/testresults/PostTestResultsNegTestTypesCancelledLvl.java [hunk 4]
```diff
@@ -118,6 +120,7 @@ public void testResultsExtraFieldTestExpiryDate() {
         testResultsSteps.validatePostErrorData("testExpiryDate", "is not allowed");
     }
 
+    @Ignore("blocked by CVSB-9639")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsNegTestTypesCancelledLvl.java]: defect was fixed, please remove the @ignore annotation

**Review by bfloreaDVSA — CHANGES_REQUESTED** [review]
please remove the @Ignore annotations

File: src/test/java/testnumber/PostTestNumber.java (hunk 2 above)
Comment (jgimber-deloitte) [reply src/test/java/testnumber/PostTestNumber.java]: Removed.

File: src/test/java/testresults/PostTestResultsNegTestTypesCancelledLvl.java (hunk 4 above)
Comment (jgimber-deloitte) [reply src/test/java/testresults/PostTestResultsNegTestTypesCancelledLvl.java]: Removed.

File: src/test/java/testresults/PostTestResultsNegTestTypesCancelledLvl.java (hunk 3 above)
Comment (jgimber-deloitte) [reply src/test/java/testresults/PostTestResultsNegTestTypesCancelledLvl.java]: Removed.

**Review by bfloreaDVSA — APPROVED** [review]

**Review by dpanzaru-deloitte — APPROVED** [review]

**Pull request merged**